_posts/2021-03-05-introducing-new-window-features.md

```diff
@@ -0,0 +1,356 @@
+    ('student_5', 10),
+    ('student_6', 20),
+    ('student_7', 16))
+SELECT student_id, result, count(*) OVER (ORDER BY result
+   RANGE BETWEEN 1 FOLLOWING AND 2 FOLLOWING) AS close_better_scores_count
+FROM students_results;
+
+ student_id | result | close_better_scores_count
+------------+--------+---------------------------
+ student_5  |     10 |                         0
+ student_7  |     16 |                         3
+ student_2  |     16 |                         3
+ student_1  |     17 |                         2
+ student_3  |     18 |                         1
+ student_4  |     18 |                         1
+ student_6  |     20 |                         0
+(7 rows)
+```
+
+Note that the frame does not contain the current row. For a particular student,
+it only includes students with better results, and not themselves. For the
+unfortunate `student_5`, there are no students with similar test results.
+Also, for the lucky `student_6` who scored the most points.
+
+![](/assets/blog/window-features/students-range.svg)
+
+Besides `ROWS` and `RANGE`, there is another way to specify the frame on
+ordered data. And yes, Trino has it! Let me introduce the second of our recent
+additions:
+
+### Support for frame type GROUPS
+
+This feature, added in
+[version 346]({{site.url}}/docs/current/release/release-346.html), allows you to
+include or exclude the whole _peer groups_ of rows in ordered data.
+
+For illustration, let's consider again the `students_results` table. For each
+student, let's find the gap between their result and the result of a student (or
+students) who did slightly better.
+
+```
+WITH students_results(student_id, result) AS (VALUES
+    ('student_1', 17),
+    ('student_2', 16),
+    ('student_3', 18),
+    ('student_4', 18),
+    ('student_5', 10),
+    ('student_6', 20),
+    ('student_7', 16))
+SELECT student_id, result, max(result) OVER (ORDER BY result
+   GROUPS BETWEEN CURRENT ROW AND 1 FOLLOWING) - result AS gap_till_better_score
+FROM students_results;
+
+ student_id | result | gap_till_better_score
+------------+--------+-----------------------
+ student_5  |     10 |                     6
+ student_7  |     16 |                     1
+ student_2  |     16 |                     1
+ student_1  |     17 |                     1
+ student_3  |     18 |                     2
+ student_4  |     18 |                     2
+ student_6  |     20 |                     0
+(7 rows)
+```
+
+The window function for each student returns the closest better result. The
+frame of type `GROUPS` used here, includes all entries equal to the current
+entry in terms of points (that is the student's _peer group_), and the next
+group.
+
+![](/assets/blog/window-features/students-groups.svg)
+
+In frame of type `GROUPS`, like in other frame types, the offset doesn't have to
+be constant. It can be any expression, as long as its type is exact numeric with
+scale 0. Simply put, we can skip integer number of groups.
```

Comment:
Simply put, we can skip any number of groups specified as an integer.

Reply:
It can be specified e.g. as decimal, too. It only has to be "effectively integer". I'll change to "we can skip any integer number of groups."